### PR TITLE
L1 events and block processing

### PR DESCRIPTION
**Pull request by badgersrus**
### Why this change is needed

[Please provide a description and a link to the underlying ticket
](https://github.com/ten-protocol/ten-internal/issues/4299)

### What changes were made as part of this PR

* Removed the `block.transactions()` loops that we used to find relevant transactions in the host and enclave:
  * `rollup_consumer.go` - looping through block transactions to find blob txs contianing rollups
  * `shared_secret_process.go` - looping through transactions to find secret initialization and requests
  * `block_message_extractor.go` - looping through receipts to find xchain messages
  * `publisher.go` - looping through block transactions to find secret responses
* Moved all of block processing logic into `blockrepository.go` where we fetch all relevant logs based on address, then filter based on the events emitted in them. When an event is found we add it to a map of the event to the list of transactions and receipts
* This map of events to transactions/ receipts is passed to the enclave in the `SubmitL1Block` where the appropriate transactions can be easily found by looking up the event type 
* Intermittent rollup duplication guarded against in the guardian when publishing the rollup takes longer than the block time 

### PR checks pre-merging

Please indicate below by ticking the checkbox that you have read and performed the required
[PR checks](https://github.com/ten-protocol/ten-internal/blob/main/dev-ops-docs/dev-pr-checks.md)

- [ ] PR checks reviewed and performed

### CODE REVIEW

**Review by tudor-malene — COMMENTED** [review]
I think it looks good in principle.

The thing that confuses me is how those `ethadapter` types fit in

File: go/common/host/services.go [hunk 1]
```diff
@@ -101,10 +101,8 @@ type L1Publisher interface {
 	InitializeSecret(attestation *common.AttestationReport, encSecret common.EncryptedSharedEnclaveSecret) error
 	// RequestSecret will send a management contract transaction to request a secret from the enclave, returning the L1 head at time of sending
 	RequestSecret(report *common.AttestationReport) (gethcommon.Hash, error)
-	// ExtractRelevantTenTransactions will return all TEN relevant tx from an L1 block
-	ExtractRelevantTenTransactions(block *types.Block, receipts types.Receipts) ([]*common.TxAndReceiptAndBlobs, []*ethadapter.L1RollupTx, []*ethadapter.L1SetImportantContractsTx)
 	// FindSecretResponseTx will return the secret response tx from an L1 block
-	FindSecretResponseTx(block *types.Block) []*ethadapter.L1RespondSecretTx
+	FindSecretResponseTx(responseTxs []*common.L1TxData) []*ethadapter.L1RespondSecretTx
```
[the remark this answers is not in the document]
Comment (badgersrus) [reply go/common/host/services.go:105]: The are pre-existing, in one of the previous versions I moved them all up into the common package which I'll do in here

File: go/common/l1_transaction.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+	AttesterSig []byte
+}
+
+type L1SetImportantContractsTx struct {
+	Key        string
+	NewAddress gethcommon.Address
+}
+
+// Sign signs the payload with a given private key
+func (l *L1RespondSecretTx) Sign(privateKey *ecdsa.PrivateKey) *L1RespondSecretTx {
```
Comment (tudor-malene) [inline go/common/l1_transaction.go:46]: this is a bit confusing here. It's probably just used during testing, so it should be moved there

**Review by tudor-malene — APPROVED** [review]
LGTM!
(minor comment)

**Comment by badgersrus** [comment]
Copying comments from the original PR which is now closed:

> Nice, I like the idea of preparing all the TEN-relevant block data on the host side and then passing it in with the block.
- I'd keep the guardian changes to a minimum, since this metadata will be strictly a function of the block being processed it makes more sense to be called at the blockrepository layer (and as you suggest, it sounds ideal for it to be in its own service or package).
- for HA hosts, the guardians currently fetch the L1 block from the block repository - I don't think it's worth cacheing/persisting the extracted data though unless it becomes an obvious hotspot, should be quick enough to process on demand for each guardian I guess? If it's accessed through the block repo service or similar though then cacheing would be easy to add.
- not clear to me either what's required to make this easily and robustly verifiable on the enclave side. Would definitely stub it out for the first PR though and hopefully we don't make something that's not easily provable...

> Actually one other concern I had, which I think is why we started doing this enclave side to begin with is that it's easy enough to prove TEN-data was found in that block, but it's not easy to prove that the host didn't miss anything.

Maybe the incentives are such that there aren't any useful attacks there. If the host misses a x-chain msg or a sequencer-permissioning message then the enclave would get into a bricked state where it could no longer process new batches but that's not v useful as an attack I imagine.

**Pull request merged**